src/mesh/parallel_mesh.C

```diff
@@ -290,10 +290,7 @@ dof_id_type ParallelMesh::parallel_max_node_id() const
 
 const Point & ParallelMesh::point (const dof_id_type i) const
```

Comment:
No, IMHO it shouldn't.

I actually thought about this!  Although I probably should have put my thoughts into a comment here rather than a git log message.

We don't want to deprecate point() for the same reason we don't want to put the shim for it into MeshBase: because some day in the distant future we may want to add mesh subclasses like CartesianMesh which generate elements and nodes on the fly, and in that case it will probably be cheaper to generate just the Point when requested, not the full Node.

We probably do want to move point() from serial_/parallel_mesh.C into unstructured_mesh.h; I just didn't do that in this PR because I was trying to keep the 3rd through Nth commits as straightforward a translation as practical.